### DISCUSSION TITLE
Upgrade to `miniflare v3`

### DISCUSSION THREAD

**Discussion by peterhirn** [nitrojs/nitro#1731] [General]
Hi,

`miniflare v3` was released quite recently (~2 months ago). As far as I understand they completely replaced the cf emulation in node.js with native `workerd` which lead to this long list of [breaking API changes](https://miniflare.dev/get-started/migrating#api-changes). Additionally there are a couple of [missing features](https://miniflare.dev/get-started/migrating#missing-features).

Mixing `miniflare v2` and `wrangler v3` in development works, but produces some unexpected behavior:

- `miniflare v2` requires additional configuration to parse config files (`.env`, `wrangler.toml`, etc.)
- state is persisted in different local folders (`.mf/` vs `.wrangler/`)

I was wondering
- is there an ongoing effort to upgrade to `miniflare v3`
- do some of the breaking changes / missing features outright prevent an upgrade
- would you be interested in a PR / how can I help

Cheers 🍺

References:
- https://github.com/unjs/nitro/pull/1251

**Comment by Hebilicious** [comment]
Hi @peterhirn, we use miniflare only for the tests, so upgrading should be relatively straightforward as we are not using a lot of the miniflare features. Feel free to open a PR.
